### PR TITLE
[GCS]Only publish fileds used by sub clients in WorkerTableData

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
Only worker id and raylet id used by sub clients in WorkerTableData, so we only publish them.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_worker_manager.cc [hunk 1]
```diff
@@ -52,8 +52,15 @@ void GcsWorkerManager::HandleReportWorkerFailure(
                      << ", address = " << worker_address.ip_address();
     } else {
       stats::UnintentionalWorkerFailures.Record(1);
+      // Only publish worker_id and raylet_id in address as they are the only fields used
```
Comment (ffbin) [inline src/ray/gcs/gcs_server/gcs_worker_manager.cc:55]: How about modify `AsyncSubscribeToWorkerFailures` synchronously, only showing `worker_id` and `raylet_id`?

**Review by rkooo567 — APPROVED** [review]
For me, the current solution looks good, but I don't have any opinion about @ffbin's request (so feel free to do so if you guys think that's better).

**Comment by WangTaoTheTonic** [comment]
I've discussed with @ffbin offline. The current optimization did not set values for other fields except raylet_it and worker_id, and it's easy to mislead other people who use sub client(they perhaps don't know what we published is delta data).
I'll try to use a different data structure in publishing, in which only that two fields be contained.

**Comment by WangTaoTheTonic** [comment]
Will ping you guys once the new solution is done! @ffbin @rkooo567

**Comment by WangTaoTheTonic** [comment]
@ffbin @rkooo567 New solution is committed. Please take a review, thanks :)

**Review by ffbin — APPROVED** [review]

**Pull request merged**